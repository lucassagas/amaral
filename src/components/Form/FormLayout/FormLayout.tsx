import { Box, Divider, Heading } from '@chakra-ui/react'
import { ReactNode } from 'react'

type FormLayoutProps = {
  title: string
  children: ReactNode
}

export function FormLayout({ children, title }: FormLayoutProps) {
  return (
    <Box flex="1" borderRadius={8} bg="gray.800" p={['6', '8']}>
      <Heading size="lg" fontWeight="normal">
        {title}
      </Heading>

      <Divider my="6" borderColor="gray.700" />

      {children}
    </Box>
  )
}
