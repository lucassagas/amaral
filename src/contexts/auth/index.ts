export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
