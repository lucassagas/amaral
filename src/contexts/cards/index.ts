export { CardContext, CardContextProvider } from './cards';

