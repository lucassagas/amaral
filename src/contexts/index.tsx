import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
