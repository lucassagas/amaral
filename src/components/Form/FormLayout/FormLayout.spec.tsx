import { AppProvider } from '@/contexts/index'
import { render } from '@testing-library/react'
import { FormLayout } from './FormLayout'

describe('<FormLayout />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <AppProvider>
        <FormLayout title="Form Example">
          <h1>Form</h1>
        </FormLayout>
      </AppProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
