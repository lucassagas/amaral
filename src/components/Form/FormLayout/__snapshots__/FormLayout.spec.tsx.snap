// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLayout /> should render correctly 1`] = `
<div>
  <div
    class="css-qqp9b6"
  >
    <h2
      class="chakra-heading css-sxvlx"
    >
      Form Example
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-e44dav"
    />
    <h1>
      Form
    </h1>
  </div>
</div>
`;
